api/types/okta.go

```diff
@@ -393,6 +410,32 @@ func (o *OktaAssignmentV1) CheckAndSetDefaults() error {
 	o.Spec.CleanupTime = o.Spec.CleanupTime.UTC()
 	o.Spec.LastTransition = o.Spec.LastTransition.UTC()
 
+	// Deduplicate targets
+	allTargets := map[OktaAssignmentTargetV1_OktaAssignmentTargetType]map[string]struct{}{}
```

Comment:
I don't disagree. So, short answer is yes, and that's handled here: https://github.com/gravitational/teleport.e/pull/2977. However, due to this issue, it's possible that there are duplicates in the backend, so it feels like this should be done somewhere at a lower level to prevent the Okta assignment processor from attempting to double assign/double remove an Okta assignment.

This logic *should* be taken care of in the Okta assignment processor as is, but just to be careful with the assignments with dupes in the backend I figured this wouldn't be a bad idea.